### PR TITLE
address #4932,fix for errors of Binding behavior

### PR DESCRIPTION
**Pull request by madneal**

### CODE REVIEW

**Review by shati-patel — COMMENTED** [review]
Thank you, @madneal! You're right about the additional error message. 

The "expression" error also appears for the second example below&mdash;would you mind updating that section too? As in:


```
/*
  Compilation errors:
  ERROR: "str" is not bound to a value.
  ERROR: expression is not bound to a value.
*/
predicate shortString(string str) {
  str.length() < 10
}
```

**Comment by madneal** [comment]
@shati-patel Sorry for the miss, added

File: docs/codeql/ql-language-reference/predicates.rst [hunk 1]
```diff
@@ -234,6 +235,7 @@ Here are a few examples of infinite predicates:
     /*
       Compilation error:
```
Comment (shati-patel) [inline docs/codeql/ql-language-reference/predicates.rst:236]: Great, thanks! One more tiny suggestion:

```suggestion
      Compilation errors:
```

Comment (madneal) [reply docs/codeql/ql-language-reference/predicates.rst]: @shati-patel fixed, thanks

**Comment by smowton** [comment]
Without the specific applicable range seen in the UI, "expression is not bound to a value" isn't very meaningful -- suggest adding `(referring to the expression i * 4)` or similar in each instance.

File: docs/codeql/ql-language-reference/predicates.rst [hunk 2]
```diff
@@ -226,14 +226,16 @@ Here are a few examples of infinite predicates:
       Compilation errors:
       ERROR: "i" is not bound to a value.
       ERROR: "result" is not bound to a value.
+      ERROR: expression is not bound to a value.
```
Comment (shati-patel) [inline docs/codeql/ql-language-reference/predicates.rst:229]: Clarify which `expression` the error refers to. (It doesn't match the exact error message in the UI, but it makes sense to add extra context for our readers 😊)

```suggestion
      ERROR: expression "i * 4" is not bound to a value.
```

File: docs/codeql/ql-language-reference/predicates.rst [hunk 3]
```diff
@@ -226,14 +226,16 @@ Here are a few examples of infinite predicates:
...
     */
     int multiplyBy4(int i) {
       result = i * 4
     }
     
     /*
-      Compilation error:
+      Compilation errors:
       ERROR: "str" is not bound to a value.
+      ERROR: expression is not bound to a value.
```
Comment (shati-patel) [inline docs/codeql/ql-language-reference/predicates.rst:238]: As above!
```suggestion
      ERROR: expression "str.length()" is not bound to a value.
```

**Review by shati-patel — COMMENTED** [review]
> Without the specific applicable range seen in the UI, "expression is not bound to a value" isn't very meaningful -- suggest adding `(referring to the expression i * 4)` or similar in each instance.

Fair point, @smowton. I've suggested another small tweak to clarify! After that, this should be ready to merge. Thank you for the help, @madneal ✨

File: docs/codeql/ql-language-reference/predicates.rst (hunk 3 above)
Comment (madneal) [reply docs/codeql/ql-language-reference/predicates.rst]: @shati-patel done

**Review by shati-patel — APPROVED** [review]
Thank you for all the edits! I'll get this merged once the tests are green 😃

**Pull request merged**